### DISCUSSION TITLE
How to use memory/conversation history correctly with agent

### DISCUSSION THREAD

**Discussion by octadion** [sinaptik-ai/pandas-ai#954] [Q&A]
How to use memory/conversation history correctly to be able to answer related to the previous question? I have used the agent and set memory size 10 but the resulting answer is not related at all.

**Comment by NM-Manasa** [comment]
+1. Do we have an example on how to use the memory/conversation memory using pandasai agents ? @gventuri

**Comment by JohnRhaenys** [comment]
@gventuri any updates on this?

**Comment by istrupin** [comment]
Would be very slick if I could also pass memory/conversation history as a parameter to the agent so I could run the agent in a way that is decoupled from the user facing application.

It seems like now the agent manages its own conversation history (which makes a lot of sense in a standalone app) but it would be helpful to be able to pass that in if I wanted to.

**Comment by gdcsinaptik** [comment]
Hi guys, please start by having a look at this documentation page: https://docs.pandas-ai.com/library#agent :)
